compiler/rustc_feature/src/accepted.rs

```diff
@@ -349,6 +349,9 @@ declare_features! (
     (accepted, repr_packed, "1.33.0", Some(33158)),
     /// Allows `#[repr(transparent)]` attribute on newtype structs.
     (accepted, repr_transparent, "1.28.0", Some(43036)),
+    /// Allows enums like Result<T, E> to be used across FFI, if T's niche value can
+    /// be used to describe E or vice-versa.
+    (accepted, result_ffi_guarantees, "1.80.0", Some(110503)),
```

Comment:
Since the previous attempt was incomplete, I think this needs a reset:
```suggestion
    (accepted, result_ffi_guarantees, "CURRENT_RUSTC_VERSION", Some(110503)),
```